File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    routes.sort(key=lambda x:x[1])
    print(routes)
    
    
    START = 0
    END = 1
    points = []
    points.append(routes[0][END])
    
    for i in range(1, len(routes)):
        if points[-1] >= routes[i][START]:
            continue
        else:
            points.append(routes[i][END])
    
    print(points)
    return len(points)

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    vector<int> points;
    sort(routes.begin(), routes.end(), [](vector<int> a, vector<int> b) -> bool {
        return a[1] < b[1];
    });
    
    points.push_back(routes[0][1]);
    for (int i = 1; i < routes.size(); i++) {
        if (points[points.size() - 1] < routes[i][0]) {
            points.push_back(routes[i][1]);
        }
    }

    return points.size();
}